roachtest: schemachange/index/tpcc/w=1000 failed

SHA: https://github.com/cockroachdb/cockroach/commits/4784fe3c51545db5fb5d411937ec1db2ef2b9761

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=schemachange/index/tpcc/w=1000 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1472753&tab=buildLog

```
The test failed on branch=provisional_201909060000_v19.2.0-beta.20190910, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190906-1472753/schemachange/index/tpcc/w=1000/run_1
	cluster.go:1735,tpcc.go:184,cluster.go:2091,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1567786968-71-n5cpu16:5 -- ./workload run tpcc --warehouses=1000 --histograms=perf/stats.json --wait=false --tolerate-errors --ramp=5m0s --duration=2h0m0s {pgurl:1-4} returned:
		stderr:
		
		stdout:
		   30014            0.0           23.4      0.0      0.0      0.0      0.0 stockLevel
		 2990.0s    37448            1.0           23.4  18253.6  18253.6  18253.6  18253.6 delivery
		 2990.0s    37448            0.0          233.6      0.0      0.0      0.0      0.0 newOrder
		 2990.0s    37448            1.0           23.4   4295.0   4295.0   4295.0   4295.0 orderStatus
		 2990.0s    37448            1.0          232.4  24696.1  24696.1  24696.1  24696.1 payment
		 2990.0s    37448            0.0           23.4      0.0      0.0      0.0      0.0 stockLevel
		 2991.0s    37448            0.0           23.4      0.0      0.0      0.0      0.0 delivery
		 2991.0s    37448            2.0          233.5  28991.0  66572.0  66572.0  66572.0 newOrder
		 2991.0s    37448            0.0           23.4      0.0      0.0      0.0      0.0 orderStatus
		 2991.0s    37448            5.0          232.3  19327.4  90194.3  90194.3  90194.3 payment
		 2991.0s    37448            0.0           23.4      0.0      0.0      0.0      0.0 stockLevel
		: signal: killed

```

SHA: https://github.com/cockroachdb/cockroach/commits/62b1678f652461bbc1aaf6bc2c0dd03105ce0ebe

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=schemachange/index/tpcc/w=1000 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1489712&tab=buildLog

```
The test failed on branch=40765, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190915-1489712/schemachange/index/tpcc/w=1000/run_1
	cluster.go:1735,tpcc.go:184,cluster.go:2091,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1568575032-75-n5cpu16:5 -- ./workload run tpcc --warehouses=1000 --histograms=perf/stats.json --wait=false --tolerate-errors --ramp=5m0s --duration=2h0m0s {pgurl:1-4} returned:
		stderr:
		
		stdout:
		2079           14.0          268.8  66572.0  81604.4 103079.2 103079.2 newOrder
		 4119.0s    22079            5.0           26.9    838.9  60129.5  60129.5  60129.5 orderStatus
		 4119.0s    22079           87.1          267.9  73014.4 103079.2 103079.2 103079.2 payment
		 4119.0s    22079           25.0           26.9  57982.1  98784.2 103079.2 103079.2 stockLevel
		E190916 01:53:45.224926 1 workload/cli/run.go:447  error in delivery: ERROR: result is ambiguous (error=rpc error: code = Unavailable desc = transport is closing [propagate]) (SQLSTATE 40003)
		 4120.0s    22103           43.9           26.9  60129.5  85899.3 103079.2 103079.2 delivery
		 4120.0s    22103           64.9          268.7  64424.5  81604.4  90194.3 103079.2 newOrder
		 4120.0s    22103            8.0           26.9     46.1  60129.5  60129.5  60129.5 orderStatus
		 4120.0s    22103          142.7          267.9  68719.5 103079.2 103079.2 103079.2 payment
		 4120.0s    22103            8.0           26.9    385.9  73014.4  73014.4  73014.4 stockLevel
		: signal: killed

```

--- a/pkg/cmd/roachtest/schemachange.go
+++ b/pkg/cmd/roachtest/schemachange.go
@@ -299,7 +299,9 @@ func makeIndexAddTpccTest(spec clusterSpec, warehouses int, length time.Duration
 		Run: func(ctx context.Context, t *test, c *cluster) {
 			runTPCC(ctx, t, c, tpccOptions{
 				Warehouses: warehouses,
-				Extra:      "--wait=false --tolerate-errors",
+				// We limit the number of workers because the default results in a lot
+				// of connections which can lead to OOM issues (see #40566).
+				Extra: fmt.Sprintf("--wait=false --tolerate-errors --workers=%d", warehouses),
 				During: func(ctx context.Context) error {
 					return runAndLogStmts(ctx, t, c, "addindex", []string{
 						`CREATE UNIQUE INDEX ON tpcc.order (o_entry_d, o_w_id, o_d_id, o_carrier_id, o_id);`,
@@ -409,7 +411,9 @@ func makeMixedSchemaChanges(spec clusterSpec, warehouses int, length time.Durati
 		Run: func(ctx context.Context, t *test, c *cluster) {
 			runTPCC(ctx, t, c, tpccOptions{
 				Warehouses: warehouses,
-				Extra:      "--wait=false --tolerate-errors",
+				// We limit the number of workers because the default results in a lot
+				// of connections which can lead to OOM issues (see #40566).
+				Extra: fmt.Sprintf("--wait=false --tolerate-errors --workers=%d", warehouses),
 				During: func(ctx context.Context) error {
 					if t.IsBuildVersion(`v19.2.0`) {
 						if err := runAndLogStmts(ctx, t, c, "mixed-schema-changes-19.2", []string{

